File: Sliding_window/Variable_Size/minwindow.js
function minSubArrayLen(arr,k){
    let minLen = Infinity;
    let windowSum =0;
    let left = 0;

    for(let right=0;right<arr.length;right++)
    {
        windowSum+=arr[right];

        while(windowSum>=k)
        {
                  minLen = Math.min(minLen,right-left+1);
            windowSum-=arr[left];
            left++;
        }

    }
    return minLen === Infinity ? 0 : minLen;

}

let arr = [2, 1,1,1, 5, 2, 3, 2];
let k = 7;
console.log(minSubArrayLen(arr, k));

File: Sliding_window/Variable_Size/minwindow.py
class Solution:
    def minWindow(self, s: str, t: str) -> str:
        if len(s) < len(t):
            return ""

        tMap = {}
        for c in t:
            tMap[c] = tMap.get(c, 0) + 1

        have, need = 0, len(tMap)
        sMap = {}

        res = [-1, -1]
        minLen = float("inf")
        left = 0

        for right in range(len(s)):
            c = s[right]
            sMap[c] = sMap.get(c, 0) + 1

            if c in tMap and sMap[c] == tMap[c]:
                have += 1

            while have == need:
                windowLen = right - left + 1

                if windowLen < minLen:
                    minLen = windowLen   # 🔥 THIS WAS MISSING
                    res = [left, right]

                leftChar = s[left]
                sMap[leftChar] -= 1

                if leftChar in tMap and sMap[leftChar] < tMap[leftChar]:
                    have -= 1

                left += 1

        l, r = res
        return s[l:r+1] if minLen != float("inf") else ""